parsers/mcmaster-parser-2.js

```diff
@@ -17,7 +16,40 @@ function convertTo24Hour(time) {
     }
     return time.replace(/(AM|PM)/, '');
 }
+/**
+ * Takes any date and turns it into the correct format, so javascript gets the right date.
+ * MM/DD/YYYY is the format the javascript object reads or YYYY/MM/DD
+ * @param {String} startDate
+ * @param {String} endDate
+ * @return {String Array} [startDate, endDate]  with format MM/DD/YYYY or YYYY/MM/DD or YYYY-MM-DD
+ */
+function fixDateFormat(sem_start, sem_end) {
+    var startDateSplit = sem_start.split("/");
+    var endDateSplit = sem_end.split("/");
 
+    var startMonth = parseInt(startDateSplit[0]);
+    var startDay = parseInt(startDateSplit[1]);
+    var startYear = parseInt(startDateSplit[2]);
+
+    var endMonth = parseInt(endDateSplit[0]);
+    var endDay = parseInt(endDateSplit[1]);
+    var endYear = parseInt(endDateSplit[2]);
+
+    var monthDiff = endMonth - startMonth;
+    var dayDiff = endDay - startDay;
+    // If it contains the '-' or the first slot is YYYY it is already in correct format
+    if (sem_start.indexOf('-') > -1 || startMonth.toString().length === 4) {
```

Comment:
I dont think I should, because what will happen is the startMonth is YYYY, is is going to be >12 and will hit the wrong condition